client/extensions/woocommerce/app/dashboard/widgets/stats-widget/queries.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+import { moment } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getUnitPeriod, getStartDate } from 'woocommerce/app/store-stats/utils';
+import { UNITS, dashboardListLimit } from 'woocommerce/app/store-stats/constants';
+
+export const getQueries = unit => {
```

Comment:
Nice! If you can make `getQueries` a function of `unit` and `dateString` and maybe move this file to `/store-stats`, I could reuse it for the other dashboard, keeping this logic all in one place.

Usage would be like:
```js
const{ orderQuery } = getQueries( unit, moment().format( 'YYYY-MM-DD' ) );
```

Reply:
I haven't done this yet but we can in a follow-up PR. Will this end up being useful for you? i.e. my referrers query only pulls 1 unit's worth of data. Wouldn't you need to pull more than that for some of your displays?

Reply:
It would be useful. But now I see what you mean. The difference is `quantity: 1` on the `referrerQuery`. Hmm...

Could we set it to `quantity: UNITS[ unit ].quantity` and override somehow if desired?